joint_trajectory_controller/test/test_trajectory_actions.cpp

```diff
@@ -197,7 +197,24 @@ class TestTrajectoryActions : public TrajectoryControllerTest
   }
 };
 
-TEST_F(TestTrajectoryActions, test_success_single_point_sendgoal)
+// From the tutorial: https://www.sandordargo.com/blog/2019/04/24/parameterized-testing-with-gtest
+class TestTrajectoryActionsTestParameterized
+: public TestTrajectoryActions,
+  public ::testing::WithParamInterface<
```

Comment:
We also did tests this way at work a few years ago when we got fascinated by the idea of tuples in C++ but you can get much more readable and maintainable code with good old structs (also sometimes referred to as named tuples ;))

Reply:
Regarding tests I'm only a copy and paste engineer :yum: I just copied it from the other test file, but if you give me a hint on how it can be improved, I can change that..